### PR TITLE
Added some UI polish for demo

### PR DESCRIPTION
**Pull request by matraj**
Final branch merge for now. Some UI tweaks and a bit of code clean up after making quick UI polish changes for the demo. Battery UI updates now based on it's status:

image

### CODE REVIEW

File: src/components/Battery.jsx [hunk 1]
```diff
@@ -18,24 +18,52 @@ class Battery extends Component {
...
 
+  _setBatteryStatus() {
+    if (this.props.charge < 10) {
+      return <ChartistGraph
+        className="Battery-bar-low"
+        data={this._getData()}
+        options={this._getOptions()}
+        type={"Bar"}
+      />;
+    } else if (this.props.charge < 20) {
```
Comment (rageandqq) [inline src/components/Battery.jsx]: You don't need "else if" after a return. This can be an if statement.

File: src/components/Battery.jsx [hunk 2]
```diff
@@ -18,24 +18,52 @@ class Battery extends Component {
...
+      data={this._getData()}
+      options={this._getOptions()}
+      type={"Bar"}
+    />;
+  }
+
+  _setBatteryStatusVal() {
+    if (this.props.charge < 10) {
+      return <span className="Battery-value-low">{this.props.charge} %</span>;
+    } else if (this.props.charge < 20) {
```
Comment (rageandqq) [inline src/components/Battery.jsx]: Same here

File: src/components/Battery.jsx [hunk 3]
```diff
@@ -18,24 +18,52 @@ class Battery extends Component {
     };
   }
 
+  _setBatteryStatus() {
```
Comment (rageandqq) [inline src/components/Battery.jsx]: This should be named `_get...`

File: src/components/Battery.jsx [hunk 4]
```diff
@@ -18,24 +18,52 @@ class Battery extends Component {
...
+    }
+    return <ChartistGraph
+      className="Battery-bar"
+      data={this._getData()}
+      options={this._getOptions()}
+      type={"Bar"}
+    />;
+  }
+
+  _setBatteryStatusVal() {
```
Comment (rageandqq) [inline src/components/Battery.jsx]: This should be named `_get...` Also maybe just `_getBatteryValue`.

File: src/components/PowerStatus.jsx [hunk 5]
```diff
@@ -1,22 +1,47 @@
...
 import '../styles/PowerStatus.css';
 
-// Create an array here that will append a new power value each iteration 
-const power = [];
+import { MAX_POWER_COMPONENTS } from '../utils/constants';
 
 class PowerStatus extends Component {
 
-  _getData() {
+  constructor() {
```
Comment (rageandqq) [inline src/components/PowerStatus.jsx]: A lot of this is on master. You will need to remove these changes. I'd recommended rebasing all the branches off of master.

File: src/styles/Battery.css [hunk 6]
```diff
@@ -25,18 +27,25 @@
   stroke: #469fe9;
 }
 
-.Battery-bar .ct-grid {
+.Battery-bar-med .ct-bar {
+  stroke-width: 42%;
```
Comment (rageandqq) [inline src/styles/Battery.css]: 42% of what? Is it of the window width? Or of what the default is?

File: src/utils/constants.js [hunk 7]
```diff
@@ -2,6 +2,7 @@ const CONSTANTS = Object.freeze({
   CAN_FRAME: 'can_frame',
   MAX_ACCEL: 4,
   MAX_SPEED: 220,
+  MAX_POWER_COMPONENTS: 7,
```
Comment (rageandqq) [inline src/utils/constants.js]: Also on master.

File: src/components/Battery.jsx (hunk 1 above)
Comment (jack-mo) [reply src/components/Battery.jsx]: This should be an `else if`
@matraj it looks like you only need to set the className in each if block, like so:
```
if (this.props.charge < 10) {
    className = "Battery-bar-low";
} else if (this.props.charge < 20) {
    className="Battery-bar-med";
} else {
    className="Battery-bar";
}
return ...
```

**Comment by rickeywang** [comment]
Wow looks a lot nicer than what I made 4 years ago. 

https://github.com/rickeywang/UWAFT-Public/tree/master/HMI

image

**Pull request merged**